starfish/core/spots/DecodeSpots/simple_lookup_decoder.py

```diff
@@ -0,0 +1,53 @@
+from typing import Dict, Tuple
+
+from starfish.core.codebook.codebook import Codebook
+from starfish.core.intensity_table.decoded_intensity_table import DecodedIntensityTable
+from starfish.core.spots.DecodeSpots.trace_builders import build_traces_sequential
+from starfish.core.types import Axes, Features, SpotFindingResults
+from ._base import DecodeSpotsAlgorithm
+
+
+class SimpleLookupDecoder(DecodeSpotsAlgorithm):
+    """
+    Decode spots by assiging the target value of a spot to the corresponding target value of the
+    round/ch it was found in. This method only makes sense to use in non mulitplexed sequential
+    assays where each r/ch pair only has one target assigned to it.
+
+    Parameters
+    ----------
+    codebook : Codebook
+        Contains codes to decode IntensityTable
+
+    """
+
+    def __init__(self, codebook: Codebook):
+        self.codebook = codebook
+
+    def run(self, spots: SpotFindingResults, *args) -> DecodedIntensityTable:
+        """
+        Decode spots by looking up the associated target value for the round and ch each spot is
+        in.
+
+        Parameters
+        ----------
+        spots: SpotFindingResults
+            A Dict of tile indices and their corresponding measured spots
+
+        Returns
+        -------
+        DecodedIntensityTable :
+            IntensityTable decoded and appended with Features.TARGET and values.
+
+        """
+        lookup_table: Dict[Tuple, str] = {}
+        for target in self.codebook[Features.TARGET]:
+            for ch_label in self.codebook[Axes.CH.value]:
+                for round_label in self.codebook[Axes.ROUND.value]:
+                    if self.codebook.loc[target, round_label, ch_label]:
+                        lookup_table[(int(round_label), int(ch_label))] = str(target.values)
```

Comment:
this seems slow, but the number of targets is probably not horrendous.

it wouldn't surprise me, however, if there were a faster way to do this.

Also, if you *are* going to do it the slow way, you might as well detect scenarios where multiple bit patterns decode to the same target.

Reply:
This is the same code we use in Codebook.to_json for saving out the codebook. I'm not sure I understand this "you might as well detect scenarios where multiple bit patterns decode to the same target." how would that make sense for a non multiplexed assay?

Reply:
If one passes in a multiplexed codebook into this decoder, it will happily decode it in probably strange and unexpected ways.

Reply:
oh yea but the description says it should only be used with non multiplexed data similarly to how the decode_per_round description says it should only be used for one hot encoded data